R/misc.R

```diff
@@ -475,39 +475,93 @@ load_regional_functional_data <- function(...) {
+      colnames(consolidated_list[[i]]) <- existing_colnames
     }
-    return(weight_db_file)
+    return(consolidated_list)
   }
-  ## Internal function to consolidate weights for a given condition
-  consolidate_weights <- function(weight_db_file, condition, variable_name_obj) {
-    # Set default for 'condition' if it is not specified
-    weights <- sapply(get_nested_element(weight_db_file[[1]],c(condition,twas_weights_table)), cbind)
-    variable_name_obj <- ifelse(is.null(variable_name_obj),"variant_names",variable_name_obj)
-    # Processing with variable_name_obj:  E=extract the specific variable object
-    variable_objs <- get_nested_element(weight_db_file[[1]],c(condition,variable_name_obj))
-    # Check and reformat variable_objs if necessary
-    variable_objs <- ifelse(grepl("^chr[0-9]+:", variable_objs), 
-                             gsub("^chr", "", variable_objs), # Remove 'chr' and convert to integer
-                             variable_objs)
-    rownames(weights) <- variable_objs
+
+  # Internal function to consolidate weights for given condition
+  consolidate_weights_list <- function(combined_all_data, conditions, variable_name_obj,twas_weights_table) {
+    # Set default for 'conditions' if they are not specified
+    if (is.null(conditions)) {
+    conditions <- names(combined_all_data)
+    }
+    combined_weights_by_condition <- lapply(conditions, function(condition) {                                         
+    sapply(get_nested_element(combined_all_data,c(condition,twas_weights_table)), cbind)
+    })
+    names(combined_weights_by_condition) <- conditions
+    if (is.null(variable_name_obj)) {
+      # Standard processing: Check for identical row numbers and consolidate
+      row_numbers <- sapply(combined_weights_by_condition, function(data) nrow(data))
+      if (length(unique(row_numbers)) > 1) {
+        stop("Not all files have the same number of rows for the specified condition.")
+      }
+      weights <- combined_weights_by_condition
+    } else {
+      # Processing with variable_name_obj: Align and merge data, fill missing with zeros
+      variable_objs <- lapply(conditions, function(condition) {
+      get_nested_element(combined_all_data,c(condition,variable_name_obj))})
+      weights <- align_and_merge(combined_weights_by_condition, variable_objs)
+    }
+    names(weights) <- conditions                        
     return(weights)
   }
 
   ## Load, validate, and consolidate data
   try({
-    weight_db_file <- validate_data(weight_db_file, condition, variable_name_obj)
-    consolidated_weights <- consolidate_weights(weight_db_file, condition, variable_name_obj)
-    return(consolidated_weights)
+    combined_all_data <- load_and_validate_data(weight_db_files, conditions, variable_name_obj)
+    weights <- consolidate_weights_list(combined_all_data, conditions, variable_name_obj,twas_weights_table)
+    return(list(combined_all_data = combined_all_data, weights = weights))
```

Comment:
@AnqiWang2021 just reminder that region values needs to be cleaned up into just that lbf matrix -- for you to worry about next PR.